### PR TITLE
ENH: raise early exception if 0d array is used in np.cross

### PR DESCRIPTION
**Pull request by DhavalParmar61**

### CODE REVIEW

**Comment by mattip** [comment]
Fixes #24079

File: numpy/core/numeric.py [hunk 1]
```diff
@@ -1581,6 +1581,10 @@ def cross(a, b, axisa=-1, axisb=-1, axisc=-1, axis=None):
         axisa, axisb, axisc = (axis,) * 3
     a = asarray(a)
     b = asarray(b)
+
+    if (a.ndim < 1) or (b.ndim < 1):
+        raise Exception("At least one array has zero dimension")
```
Comment (mattip) [inline numpy/core/numeric.py:1586]: If you want to catch a ValueError, you need to use ValueError as an exception. Also, there is still a grammar error in the message.
```suggestion
        raise ValueError("At least one array has zero dimensions")
```

**Comment by DhavalParmar61** [comment]
@mattip Made necessary changes.

**Comment by mattip** [comment]
Thanks @DhavalParmar61

**Comment by charris** [comment]
Lesson from this: don't merge main for something that might be backported :) Or if you do, squash commit the result.

**Comment by mattip** [comment]
Sorry, I didn't consider the implications of merging this vis-a-vis backporting.

**Pull request merged**